source/fab/target/zero.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+
+"""
+Class which implements a zero-configuration build target.
+"""
+
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
+
+from .base import FabTargetBase
+from ..tools import Category, ToolBox, ToolRepository
+from ..build_config import BuildConfig
+from ..steps.grab.folder import grab_folder
+from ..steps.find_source_files import find_source_files
+from ..steps.preprocess import preprocess_fortran
+from ..steps.c_pragma_injector import c_pragma_injector
+from ..steps.analyse import analyse
+from ..steps.compile_fortran import compile_fortran
+from ..steps.compile_c import compile_c
+from ..steps.link import link_exe
+
+
+class FabZeroConfig(FabTargetBase):
+    """Implementation of a zero-configuration build target."""
+
+    project_name = "zero-config"
+
+    @staticmethod
+    def add_arguments(parser: ArgumentParser) -> None:
+        """Add zero-config mode command line options.
+
+        Zero configuration requires the name of the directory
+        containing the source code.
+
+        :param ArgumentParser parser: an existing command line argument parser.
+        """
+
+        parser.add_argument(
+            "source", type=Path, nargs="?", default=".", help="source directory"
+        )
+
+    @staticmethod
+    def check_arguments(parser: ArgumentParser, args: Namespace) -> None:
```

Comment:
I don't really see how this class does a zero-config build. It contains three methods, that are not called. The check argument is static, so how can a derived class overwrite this to add its own validity tests?
I am probably missing something, but this PR feels incomplete. It adds some files without actually using it (short of the testing). Passing the parser in as argument doesn't feel very OO. Shouldn't the class have a parser? Then a derived class can add its own arguments etc.